server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1408,7 +1408,8 @@ public void loadIntoContext(ShardSearchRequest request, SearchContext context, Q
         final DirectoryReader directoryReader = context.searcher().getDirectoryReader();
 
         boolean[] loadedFromCache = new boolean[] { true };
-        BytesReference bytesReference = cacheShardLevelResult(context.indexShard(), directoryReader, request.cacheKey(),
+        BytesReference cacheKey = request.cacheKey(context.getQueryShardContext().mappingKey());
```

Comment:
The cache key is serialized so we have to serialize *something*. We could use the hash code of the lookup but it when we talked about his in a group a week ago we said we just didn't want to have to think about the chance of collisions.

Reply:
I talked with @jtibshirani and she pointed out that we *don't* have to serialize something. She totally said that 13 hours ago but I didn't understand because I was being dense. I'm going to try and do the thing she proposed.

Reply:
And done!